cmd/cert-checker/main.go

```diff
@@ -0,0 +1,259 @@
+// Copyright 2015 ISRG.  All rights reserved
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package main
+
+import (
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"runtime"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/cactus/go-statsd-client/statsd"
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/codegangsta/cli"
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/jmhodges/clock"
+	gorp "github.com/letsencrypt/boulder/Godeps/_workspace/src/gopkg.in/gorp.v1"
+
+	"github.com/letsencrypt/boulder/cmd"
+	"github.com/letsencrypt/boulder/core"
+	blog "github.com/letsencrypt/boulder/log"
+	"github.com/letsencrypt/boulder/policy"
+	"github.com/letsencrypt/boulder/sa"
+)
+
+const (
+	good = "valid"
+	bad  = "invalid"
+
+	filenameLayout = "20060102"
+
+	checkPeriod = time.Hour * 24 * 90
+
+	batchSize = 1000
+)
+
+type report struct {
+	GoodCerts int64
+	BadCerts  int64
+	Entries   map[string]reportEntry
+}
+
+func (r *report) save(directory string) error {
+	now := time.Now()
```

Comment:
The report object should have fields `begin` and `end`, populated by the caller at the time the run begins. For instance, if a run begins late on one day, but ends on the next day, we want the filename to reflect that properly.